.travis.yml

```diff
@@ -25,6 +25,16 @@ jobs:
       deploy: skip
       after_success: skip
       after_failure: skip
+    - stage: 'style checks'
+      name: 'mypy'
+      language: 'python'
+      python: '3.6'
+      install: python3 -m pip install 'mypy' --force-reinstall
+      script: mypy --config-file=src/bindings/python/.mypy.ini
```

Comment:
Does this run on the entire codebase (i.e., does it include `src/cmd`), or just on the bindings?  If the former, can we have it run on the front-end commands too?  If the latter, do we want to create a script like `./scripts/pylint` for mypy, to make running locally easier? (not that this is hard to remember, but maybe having all the linters and formatters in the same place will make it easier to remember to run).

Reply:
Let me check something, if I move the mypy.ini to the root it should just be "run mypy with no args" which would make it easier to cover the whole project, I'll do one of the two in a commit in a bit.

Reply:
mypy.ini is now at the root, just run `mypy` to check essentially everything.